OrbitGl/LiveFunctionsController.cpp

```diff
@@ -207,6 +208,50 @@ void LiveFunctionsController::AddIterator(FunctionInfo* function) {
   Move();
 }
 
+void LiveFunctionsController::AddFrameTrack(const FunctionInfo& function) {
+  auto function_address = FunctionUtils::GetAbsoluteAddress(function);
+  std::vector<std::shared_ptr<TimerChain>> chains =
+      GCurrentTimeGraph->GetAllThreadTrackTimerChains();
+
+  std::vector<uint64_t> all_start_times;
+
+  for (auto& chain : chains) {
+    if (!chain) continue;
+    for (TimerChainIterator it = chain->begin(); it != chain->end(); ++it) {
+      TimerBlock& block = *it;
+      for (uint64_t i = 0; i < block.size(); i++) {
+        TextBox& box = block[i];
+        if (box.GetTimerInfo().function_address() == function_address) {
+          all_start_times.push_back(box.GetTimerInfo().start());
+        }
+      }
+    }
+  }
+  std::sort(all_start_times.begin(), all_start_times.end());
+
+  for (size_t k = 0; k < all_start_times.size() - 1; ++k) {
+    TimerInfo frame_timer;
+
+    // TID is meaningless for this timer (start and end can be on two different threads).
+    constexpr const int32_t kUnusedThreadId = -1;
+    frame_timer.set_thread_id(kUnusedThreadId);
+    frame_timer.set_start(all_start_times[k]);
+    frame_timer.set_end(all_start_times[k + 1]);
+    // Depth is used to indicate i the timer is at an even or odd position in the sequence
+    // of timers. This influences the coloring of the timer. While this depends on other
+    // data (order of timers here), for rendering this information needs to be
+    // available locally for the timer and hence is stored in the timer data structure.
+    frame_timer.set_depth(k % 2);
```

Comment:
Suggestion: Maybe we could use the user_data field instead of "depth" and store the "index" (k) of the timer? In the tooltip, we could then add "this is the k-th frame". Then we could also use the index to choose the draw color.

Reply:
That's a great suggestion, will do that.

Reply:
Done.